src/Cache/Engine/RedisEngine.php

```diff
@@ -228,14 +228,27 @@ public function clear($check)
         if ($check) {
             return true;
         }
-        $keys = $this->_Redis->getKeys($this->_config['prefix'] . '*');
 
-        $result = [];
-        foreach ($keys as $key) {
-            $result[] = $this->_Redis->del($key) > 0;
+        $this->_Redis->setOption(Redis::OPT_SCAN, Redis::SCAN_RETRY);
```

Comment:
This is causing type errors in 4.x  https://travis-ci.org/cakephp/cakephp/jobs/567855601#L822.
I am not sure what version of redis/predis travis is using. But even it only occurs for older version it could be an issue for our users.

Reply:
So the issue has been https://github.com/phpredis/phpredis/pull/1540. Not sure in which version.

Reply:
Fix is available in 5.0.0. So essentially one won't be able to used Redis 4.x with Cake 4.0.

Reply:
Do `(string)Redis::SCAN_RETRY` typecast would help?

Reply:
> So essentially one won't be able to use Redis 4.x with Cake 4.0.

You mean the client library cannot be 4.x, not that you cannot use Redis 4.x.

Reply:
@garas Maybe, I haven't checked.

@josegonzalez Yes, I should have worded it better.

Reply:
Typecasting to string worked https://github.com/cakephp/cakephp/pull/13469/files#diff-61e014e894b3dcbac9bd2425ec47490bR237